### PR TITLE
refactor: resolve feedback for problem report

### PR DESCRIPTION
**Pull request by nbAmit**
Resolve the feedback #560 

Signed-off-by: Amit Padmani <amit@northernblock.io>

### CODE REVIEW

File: packages/core/src/modules/connections/services/ConnectionService.ts [hunk 1]
```diff
@@ -433,8 +442,8 @@ export class ConnectionService {
       )
     }
 
-    connectionRecord.errorMsg = `${connectionProblemReportMessage.description.code} : ${connectionProblemReportMessage.description.en}`
-    await this.updateState(connectionRecord, ConnectionState.None)
+    connectionRecord.errorMessage = `${connectionProblemReportMessage.description.code} : ${connectionProblemReportMessage.description.en}`
+    await this.updateState(connectionRecord, connectionRecord.state)
```
Comment (TimoGlastra) [inline packages/core/src/modules/connections/services/ConnectionService.ts:446]: The state won't change, so I think we should just call the connection repository with update directly as this will emit a state changed event.

Comment (TimoGlastra) [reply packages/core/src/modules/connections/services/ConnectionService.ts:446]: Also, with the `None` state being removed, there is no way to determine that you received a problem report, so we should probably look into adding some events related to problem reports in follow up PRs

File: packages/core/src/modules/credentials/services/CredentialService.ts [hunk 2]
```diff
@@ -743,8 +745,8 @@ export class CredentialService {
     )
 
     // Update record
-    credentialRecord.errorMsg = `${credentialProblemReportMessage.description.code}: ${credentialProblemReportMessage.description.en}`
-    await this.updateState(credentialRecord, CredentialState.None)
+    credentialRecord.errorMessage = `${credentialProblemReportMessage.description.code}: ${credentialProblemReportMessage.description.en}`
+    await this.updateState(credentialRecord, credentialRecord.state)
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/services/CredentialService.ts:749]: Same here

File: packages/core/tests/helpers.ts [hunk 3]
```diff
@@ -93,6 +93,7 @@ export async function waitForProofRecord(
     state?: ProofState
     previousState?: ProofState | null
     timeoutMs?: number
+    errorMessage?: string
```
Comment (TimoGlastra) [inline packages/core/tests/helpers.ts:96]: You also need to add a `filter` to the `waitForProofRecordSubject` and `waitForCredentialRecordSubject` methods for the `errorMessage` param to be taken into account

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
Nice! Really great improvements :)

Few nitpicks

File: packages/core/tests/helpers.ts (hunk 3 above)
Comment (nbAmit) [reply packages/core/tests/helpers.ts:96]: Sorry @TimoGlastra, I didn't get you.
Can you please explain in detail?

Comment (TimoGlastra) [reply packages/core/tests/helpers.ts:96]: Sorry, let me rephrase.

Adding the `errorMessage` to the parameters of `waitForProofRecord` doesn't add `errorMessage` as a filter to the `waitForProofRecord` by itself. So passing `errorMessage` to the `waitForProofRecord` (as you do in the tests) doesn't do anything.

You would need to add an extra line to `waitForProofRecordSubject`:

```ts
.filter((e) => errorMessage === undefined || e.payload.proofRecord.errorMessage === errorMessage),
```

same for credentials

File: packages/core/src/modules/connections/services/ConnectionService.ts (hunk 1 above)
Comment (nbAmit) [reply packages/core/src/modules/connections/services/ConnectionService.ts:446]: Hi @TimoGlastra, If we update the `updateState` with an `update`, it will be difficult to capture the event in the e2e test case?

https://github.com/hyperledger/aries-framework-javascript/blob/1041aa77bdc1c441d99e7852d4062766520d6597/packages/core/tests/credentials.test.ts#L483

File: packages/core/src/modules/credentials/services/CredentialService.ts (hunk 2 above)
Comment (nbAmit) [reply packages/core/src/modules/credentials/services/CredentialService.ts:749]: Hi @TimoGlastra, If we update the `updateState` with an `update`, it will be difficult to capture the event in the e2e test case?

https://github.com/hyperledger/aries-framework-javascript/blob/1041aa77bdc1c441d99e7852d4062766520d6597/packages/core/tests/credentials.test.ts#L483

Comment (TimoGlastra) [reply packages/core/src/modules/credentials/services/CredentialService.ts:749]: Yes, see my comment above:

> Also, with the None state being removed, there is no way to determine that you received a problem report, so we should probably look into adding some events related to problem reports in follow up PRs


We should probably add a new event (e.g. `ConnectionProblemReportReceived`, `CredentialProblemReportReceived` etc...) where the event looks something like:

```ts
{
   payload: {
     credentialRecord: { /* the record */ },
     problemReportMessage: { /* the problem report message */ }
}
}
```

Comment (TimoGlastra) [reply packages/core/src/modules/credentials/services/CredentialService.ts:749]: You can either add it to the current PR, or you can bypass it by listening for the `AgentMessageProcessedEvent` for now and filtering on the problem report @type

Comment (nbAmit) [reply packages/core/src/modules/credentials/services/CredentialService.ts]: @TimoGlastra , As of now I have removed the test case and I will add it into upcoming PR with the problem-report events

**Review by TimoGlastra — APPROVED** [review]
LGTM @nbAmit
